### PR TITLE
[vcpkg] json format for search

### PR DESCRIPTION
**Pull request by dan-shaw**

### CODE REVIEW

File: toolsrc/src/vcpkg/commands.search.cpp [hunk 1]
```diff
@@ -60,8 +80,9 @@ namespace vcpkg::Commands::Search
         }
     }
 
-    static constexpr std::array<CommandSwitch, 1> SEARCH_SWITCHES = {{
+    static constexpr std::array<CommandSwitch, 2> SEARCH_SWITCHES = {{
         {OPTION_FULLDESC, "Do not truncate long text"},
+        {OPTION_JSON, "List libraries in JSON format"},
```
Comment (ras0219-msft) [inline toolsrc/src/vcpkg/commands.search.cpp:85]:
```suggestion
        {OPTION_JSON, "(experimental) List libraries in JSON format"},
```

File: toolsrc/src/vcpkg/commands.search.cpp [hunk 2]
```diff
@@ -16,7 +16,27 @@ using vcpkg::PortFileProvider::PathsPortFileProvider;
 namespace vcpkg::Commands::Search
```
Comment (ras0219-msft) [inline toolsrc/src/vcpkg/commands.search.cpp:16]: We also should ensure that we do not emit the survey message when passing `x-json`.

Comment (dan-shaw) [reply toolsrc/src/vcpkg/commands.search.cpp:14]: I don't think vcpkg runs the survey message when using the search command.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**